vcd/datasource_nsxt_tier0_router.go

```diff
@@ -0,0 +1,82 @@
+package vcd
+
+import (
+	"fmt"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func datasourceVcdNsxtTier0Router() *schema.Resource {
+	return &schema.Resource{
+		Read: datasourceNsxtTier0RouterRead,
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of NSX-T Tier-0 router.",
+			},
+			"nsxt_manager_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "ID of NSX-T manager.",
+			},
+			"is_assigned": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Defines if Tier-0 router is already assigned to external network.",
+			},
+		},
+	}
+}
+
+// datasourceNsxtTier0RouterRead has special behavior. By default `GetImportableNsxtTier0RouterByName` which uses API
+// endpoint `1.0.0/nsxTResources/importableTier0Routers` does not return Tier-0 routers when they are used in external
+// networks. This causes a problem in regular Terraform flow - when user uses this datasource to reference Tier-0 router
+// for external network creation - next "apply" would fail with "Tier 0 router not found error". If original endpoint
+// does not find Tier-0 router - then this datasource queries all defined external networks and looks for Tier-0 router
+// backing by name.
+func datasourceNsxtTier0RouterRead(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	nsxtManagerId := d.Get("nsxt_manager_id").(string)
+	tier0RouterName := d.Get("name").(string)
+
+	tier0Router, err := vcdClient.GetImportableNsxtTier0RouterByName(tier0RouterName, nsxtManagerId)
+	if !govcd.ContainsNotFound(err) && err != nil {
```

Comment:
```suggestion
	if err != nil  && !govcd.ContainsNotFound(err) {
```

Reply:
I flipped it but this doesn't matter as `govcd.ContainsNotFound` checks if err != nil

Reply:
thank you, just readability, maybe personal feeling :)